models/src/main/scala/it/pagopa/interop/agreementmanagement/model/agreement/PersistentAgreement.scala

```diff
@@ -14,6 +14,7 @@ final case class PersistentAgreement(
   verifiedAttributes: Seq[PersistentVerifiedAttribute],
   suspendedByConsumer: Option[Boolean],
   suspendedByProducer: Option[Boolean],
+  suspendedByPlatform: Option[Boolean],
```

Comment:
Is it worth considering to remodel this thing as a List[Enum]? As a later refactor, maybe.

Reply:
yes, it could be a nice improvement. 
Just 2 cents: imo an advantage of the current solution is that it allows to see at a glance if someone has revoked and and then re-enabled to resource (because the option would be a `Some(false)`